activestorage/test/models/preview_test.rb

```diff
@@ -70,6 +70,61 @@ class ActiveStorage::PreviewTest < ActiveSupport::TestCase
     assert_changes -> { !!preview.image.variant(transformations)&.send(:processed?) }, to: true do
       preview.processed
     end
+
+    image = read_image(preview.image.variant(transformations))
+    assert_equal 216, image.width
+    assert_equal 280, image.height
+  end
+
+  test "#processed does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed
+    end
+  end
+
+  test "#download does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed.download {  }
+    end
+  end
+
+  test "#key does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed.key
+    end
+  end
+
+  test "#key #download and #url raise `UnprocessedError` when preview is not processed" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    preview = blob.preview({})
+
+    %i[key url download].each do |meth|
+      assert_raises ActiveStorage::Preview::UnprocessedError do
+        preview.send(meth)
+      end
+    end
+  end
+
+  test "#url does not generate a variant when provided an empty transformation" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    empty_transformation = {}
+    preview = blob.preview(empty_transformation)
+
+    assert_no_changes -> { !!preview.image.variant(empty_transformation)&.send(:processed?) } do
+      preview.processed.url
```

Comment:
It might be good to also assert that the `preview.url` is equal to `blob.preview_image.url`.

Reply:
I agree